common/common/src/test/java/io/helidon/build/common/FileUtilsTest.java

```diff
@@ -0,0 +1,188 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.build.common;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+
+import static io.helidon.build.common.FileUtils.deleteDirectoryContent;
+import static io.helidon.build.common.FileUtils.isPosix;
+import static io.helidon.build.common.FileUtils.list;
+import static io.helidon.build.common.FileUtils.newZipFileSystem;
+import static io.helidon.build.common.FileUtils.unzip;
+import static io.helidon.build.common.FileUtils.zip;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static java.nio.file.attribute.PosixFilePermission.GROUP_EXECUTE;
+import static java.nio.file.attribute.PosixFilePermission.OTHERS_EXECUTE;
+import static java.nio.file.attribute.PosixFilePermission.OWNER_EXECUTE;
+import static java.nio.file.attribute.PosixFilePermission.OWNER_READ;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class FileUtilsTest {
+
+    private static Path zipDir;
+    private static Path outputDir;
+    private static Path zipFile;
+
+    @BeforeAll
+    static void setup() throws IOException {
+        outputDir = Files.createTempDirectory("output");
+        zipFile = createZip();
+        zipDir = Files.createTempDirectory("zip");
+        Files.createDirectories(zipDir.resolve("dir"));
+        Files.createFile(zipDir.resolve("file1"));
+        Files.createFile(zipDir.resolve("dir/file2"));
+    }
+
+    private static Path createZip() throws IOException {
+        Path zip = Files.createTempDirectory("").resolve("archive.zip");
+        Path source = Files.createTempFile("file", "");
+        try (FileSystem fs = newZipFileSystem(zip)) {
+            Path target = fs.getPath("file");
+            Files.copy(source, target, REPLACE_EXISTING);
+            Files.setPosixFilePermissions(target, Set.of(OWNER_READ, OWNER_EXECUTE));
+        }
+        return zip;
+    }
+
+    @AfterEach
+    void clean() throws IOException {
+        deleteDirectoryContent(outputDir);
```

Comment:
Can avoid doing this ? This will make it hard to troubleshoot.
How about using a unique directory as under target for the output directory ?

See `io.helidon.build.common.test.utils.TestFiles.targetDir(Class<?>)`